mls-rs-uniffi/src/lib.rs

```diff
@@ -0,0 +1,524 @@
+    pub async fn create_group(&self, group_id: Option<Vec<u8>>) -> Result<Group, Error> {
+        let extensions = mls_rs::ExtensionList::new();
+        let inner = match group_id {
+            Some(group_id) => {
+                self.inner
+                    .create_group_with_id(group_id, extensions)
+                    .await?
+            }
+            None => self.inner.create_group(extensions).await?,
+        };
+        Ok(Group {
+            inner: Arc::new(Mutex::new(inner)),
+        })
+    }
+
+    /// Join an existing group.
+    ///
+    /// See [`mls_rs::Client::join_group`] for details.
+    pub async fn join_group(&self, welcome_message: Arc<Message>) -> Result<JoinInfo, Error> {
+        let welcome_message = arc_unwrap_or_clone(welcome_message);
+        let (group, new_member_info) = self.inner.join_group(None, welcome_message.inner).await?;
+
+        let group = Arc::new(Group {
+            inner: Arc::new(Mutex::new(group)),
+        });
+        let group_info_extensions = Arc::new(new_member_info.group_info_extensions.into());
+        Ok(JoinInfo {
+            group,
+            group_info_extensions,
+        })
+    }
+}
+
+#[derive(Clone, Debug, uniffi::Object)]
+pub struct CommitOutput {
+    _inner: mls_rs::group::CommitOutput,
+}
+
+impl From<mls_rs::group::CommitOutput> for CommitOutput {
+    fn from(inner: mls_rs::group::CommitOutput) -> Self {
+        Self { _inner: inner }
+    }
+}
+
+#[derive(Clone, Debug, uniffi::Object)]
+pub struct SigningIdentity {
+    inner: mls_rs_core::identity::SigningIdentity,
+}
+
+impl From<mls_rs_core::identity::SigningIdentity> for SigningIdentity {
+    fn from(inner: mls_rs_core::identity::SigningIdentity) -> Self {
+        Self { inner }
+    }
+}
+
+/// An MLS end-to-end encrypted group.
+///
+/// The group is used to send and process incoming messages and to
+/// add/remove users.
+///
+/// See [`mls_rs::Group`] for details.
+#[derive(Clone, uniffi::Object)]
+pub struct Group {
+    inner: Arc<Mutex<mls_rs::Group<Config>>>,
+}
+
+impl Group {
+    async fn inner(&self) -> tokio::sync::MutexGuard<'_, mls_rs::Group<Config>> {
+        self.inner.lock().await
+    }
+
+    async fn index_to_identity(
+        &self,
+        index: u32,
+    ) -> Result<mls_rs_core::identity::SigningIdentity, Error> {
+        let group = self.inner().await;
+        let member = group
+            .member_at_index(index)
+            .ok_or(MlsError::InvalidNodeIndex(index))?;
+        Ok(member.signing_identity)
+    }
+}
+
+/// Extract the basic credential identifier from a  from a key package.
+fn signing_identity_to_identifier(
+    signing_identity: &mls_rs_core::identity::SigningIdentity,
+) -> Result<Vec<u8>, Error> {
+    match &signing_identity.credential {
+        identity::Credential::Basic(credential) => Ok(credential.identifier.clone()),
+        _ => Err(MlsError::RequiredCredentialNotFound(BasicCredential::credential_type()).into()),
+    }
+}
```

Comment:
Suggestion: the same functionality is achieved by `return BasicIdentityProvider::new().identity(&signing_identity)` with `use mls_rs_core::identity::IdentityProvider`